src/vs/workbench/services/url/browser/urlService.ts

```diff
@@ -43,13 +45,29 @@ export class BrowserURLService extends AbstractURLService {
 	private provider: IURLCallbackProvider | undefined;
 
 	constructor(
-		@IWorkbenchEnvironmentService environmentService: IWorkbenchEnvironmentService
+		@IWorkbenchEnvironmentService environmentService: IWorkbenchEnvironmentService,
+		@IOpenerService openerService: IOpenerService,
+		@IProductService productService: IProductService
 	) {
 		super();
 
 		this.provider = environmentService.options?.urlCallbackProvider;
-
 		this.registerListeners();
+
+		const that = this;
+		openerService.registerOpener({
```

Comment:
`registerOpener` returns a disposable so I would register it (`this._register`)

Reply:
Great catch.